tests/unit_tests/test_recipe_parser.py

```diff
@@ -18,9 +19,10 @@
 @pytest.mark.parametrize("recipe_test_data", test_cases)
 @pytest.mark.asyncio
 async def test_recipe_parser(recipe_test_data: RecipeSiteTestCase):
-    recipe, _ = await scraper.create_from_url(recipe_test_data.url)
+    translator = local_provider()
+    recipe, _ = await scraper.create_from_url(recipe_test_data.url, translator)
 
     assert recipe.slug == recipe_test_data.expected_slug
-    assert len(recipe.recipe_instructions) == recipe_test_data.num_steps
+    assert len(recipe.recipe_instructions or []) == recipe_test_data.num_steps
```

Comment:
@michael-genson this is syntax I'm not familiar with.. is this ultimately saying "give me the length of `recipe.recipe_instructions` if it is defined, or the length of the array `[]` (i.e. 0) if it isn't"?

Reply:
Yup, that's correct. If `recipe.recipe_instructions` is `None`, `len(None)` will fail. Another more verbose way of writing it in one line:
```python
assert len(recipe.recipe_instructions) if recipe.recipe_instructions else 0 == recipe_test_data.num_steps
```

Or, more clearly in a few lines:
```python
if recipe.recipe_instructions:
    assert len(recipe.recipe_instructions) == recipe_test_data.num_steps
else:
    assert recipe_test_data.num_steps == 0  # or assert not recipe_test_data.num_steps
```

But the `or` syntax is super nice, and easy to read once you're familiar with it (it threw me for a loop the first time I saw it, too)